bundles/wps/wps-service-scripts/src/test/java/org/orbisgis/wpsservicescripts/OrbisGISWpsScriptPluginTest.java

```diff
@@ -0,0 +1,235 @@
+ * version.
+ *
+ * OrbisGIS is distributed in the hope that it will be useful, but WITHOUT ANY
+ * WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR
+ * A PARTICULAR PURPOSE. See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * OrbisGIS. If not, see <http://www.gnu.org/licenses/>.
+ *
+ * For more information, please consult: <http://www.orbisgis.org/>
+ * or contact directly:
+ * info_at_ orbisgis.org
+ */
+package org.orbisgis.wpsservicescripts;
+
+import junit.framework.Assert;
+import net.opengis.ows._2.CodeType;
+import net.opengis.wps._2_0.*;
+import net.opengis.wps._2_0.GetCapabilitiesType;
+import org.junit.Test;
+import org.orbisgis.frameworkapi.CoreWorkspace;
+import org.orbisgis.wpsservice.LocalWpsServer;
+import org.orbisgis.wpsservice.controller.process.ProcessIdentifier;
+import org.orbisgis.wpsservice.model.DataType;
+
+import java.beans.PropertyChangeListener;
+import java.io.*;
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+
+/**
+ * Class to test the well working of the plugin life cycle :
+ *
+ * On activating : add all the script of the resource folder to the WPS server.
+ *
+ * On deactivating : remove all the script add on activating.
+ *
+ * @author Sylvain PALOMINOS
+ */
+public class OrbisGISWpsScriptPluginTest {
+
+    /**
+     * Test the life cycle of the plugin.
+     */
+    @Test
+    public void testPluginLifeCycle(){
+        //Gets the list of the script files add by the plugin
+        CustomWpsService localWpsServer = new CustomWpsService();
+        CustomCoreWorkspace customCoreWorkspace = new CustomCoreWorkspace();
+        Assert.assertNotNull("Unable to create the temporary folder for the custom CoreWorkspace.",
+                customCoreWorkspace.getApplicationFolder());
+        OrbisGISWpsScriptPlugin plugin = new OrbisGISWpsScriptPlugin();
+        plugin.setLocalWpsService(localWpsServer);
+        plugin.setCoreWorkspace(new CustomCoreWorkspace());
+        plugin.activate();
+        List<File> addScriptList = localWpsServer.getAddScriptList();
+        //Gets the list of the script files contained in the resource folder of the plugin
+        File folder = new File(this.getClass().getResource("scripts").getFile());
+        List<File> resourceGroovyScriptList = getAllGroovyScripts(folder);
+        //Test if the resource files are add by the plugin
+        for(File resourceScript : resourceGroovyScriptList){
+            boolean isResourceScriptAdd = false;
+            for(File addScript : addScriptList){
+                if(resourceScript.getName().equals(addScript.getName())){
+                    isResourceScriptAdd = true;
+                }
+            }
+            Assert.assertTrue("The resource file '"+resourceScript.getName()+"' should be add by the plugin.", isResourceScriptAdd);
```

Comment:
It tests the OrbisGISWpsPlugin so it loads the scripts of the bundle wps-service-scripts.